### DISCUSSION TITLE
Websocket testing on ASGI app results in "The future belongs to a different loop" error

### DISCUSSION THREAD

**Discussion by roysagi** [frankie567/httpx-ws#32] [Q&A]
I am currently working on an app based on fastapi. Up until now the app only supported HTTP calls and I am now trying to add a websocket call which is based on a different HTTP call, both calls use the same dependencies.

The HTTP call works fine in testing but for some reason the websocket call is constantly getting this error in unit tests:
```
 httpx_ws._api.WebSocketDisconnect: (<CloseReason.INTERNAL_ERROR: 1011>, 'The future belongs to a different loop than the one specified as the loop argument')
```

I am guessing the test client is using a different loop than the app which results in this error for some reason (?)
Any way I can set the loop used by the test client? or any other means to fix this issue?

here is my test code:
```
async def test_websocket(
     test_api_user_token, app
):
    async with AsyncClient(
        app=app,
        transport=ASGIWebSocketTransport(app, raise_app_exceptions=False),
        base_url="https://testserver",
    ) as client:
        client.headers = {"Authorization": f"Bearer {test_api_user_token}"}

        async with aconnect_ws(f"ws://testserver/ws", client) as websocket:
            await asyncio.sleep(3)
```

Any help here would be appreciated.

**Comment by frankie567** [comment]
Two things to check:

1. Don't set the `app` argument on `AsyncClient`, only `transport` argument is necessary
2. I suppose you use [`pytest-asyncio`](https://pypi.org/project/pytest-asyncio/) to run async tests. Make sure you define the `event_loop` fixture as described here: https://pytest-asyncio.readthedocs.io/en/latest/reference/fixtures.html#fixtures
